src/tags/header.rs

```diff
@@ -0,0 +1,970 @@
+//! Type definitions for "multiboot2 header tags". These tags occur in a binary if it
+//! is multiboot2-compliant, for example a kernel.
+//!
+//! The values are taken from the example C code at the end of the official multiboot2 spec.
+//!
+//! This type definitions are only beneficial to parse such a structure. They can't be used
+//! to construct a multiboot2 header as a static global variable. To write a multiboot2
+//! header, it is highly recommended to do this directly in assembly, because of the
+//! unknown size of structs or some addresses are not known yet (keyword: relocations).
+
+// the names speak for themselves in most cases
+#![allow(missing_docs)]
+
+use alloc::vec::Vec;
+use core::fmt;
+use core::fmt::{Debug, Formatter};
+use core::mem::size_of;
+use tags::mbi::TagType;
+use core::marker::PhantomData;
+
+
+/// Value must be present in multiboot2 header.
+pub const MULTIBOOT2_HEADER_MAGIC: u32 = 0xe85250d6;
+
+pub unsafe fn load_mb2_header<'a>(addr: usize) -> Multiboot2Header<'a> {
```

Comment:
Why not as a method on `Multiboot2Header`?

Reply:
I wanted to structure the library similar to the existing code. But yeah, good idea.